### DISCUSSION TITLE
Using georeferenced maps in mapsforge

### DISCUSSION THREAD

**Discussion by AFA141** [mapsforge/mapsforge#1308] [Q&A] [answered]
Hello, i am developing my application and i am using mapsforge framework. I want to make georeferenced map from some image for example from jpg file. And i want to use that georeferenced map in my application but framework can work only with .map format. What is the best and easiest way to make georeferenced map and import in my application? Thanks.

**Comment by devemux86** [comment]
You can use [VTM map library](https://github.com/mapsforge/vtm) which supports raster tiles in MBTiles databases.
(see the [MBTilesBitmapActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/MBTilesBitmapActivity.java) sample)

However databases do not work with file streams on Android scoped storage.

**Reply by AFA141** [reply]
Okay so i have to change to VTM library. Does VTM have all functionalities as Mapsforge ? I will use application on Desktop only.

**Comment by devemux86** [comment]
> Does VTM have all functionalities as Mapsforge ?

VTM is an OpenGL map library, so it has more features and is compatible with Mapsforge maps.

However the MBTiles databases are implemented (by community) only on Android (not desktop).

**Reply by AFA141** [reply]
My mistake, I should type in begin that I use maps only for Desktop application. What i can use instead of MBTiles?

**Comment by devemux86** [comment, marked as answer]
> I should type in begin that I use maps only for Desktop application. What i can use instead of MBTiles?

You need to implement the reader for map databases on desktop.
For example we use the [xerial/sqlite-jdbc](https://github.com/xerial/sqlite-jdbc) lib for the POI on desktop.

Or in Mapsforge you could try TMS tiles via a prefilled cache with raster tiles.
(see the [TileStoreLayerViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/TileStoreLayerViewer.java) sample and the [TMS sample tiles](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-samples-android/data/tilestore))

**Reply by AFA141** [reply]
I will try with TMS tiles, this samples is placed in android part? Can i use android samples in desktop too?

**Comment by devemux86** [comment]
> Can i use android samples in desktop too?

If the code exists in the core map modules, you can try it on both platforms.
https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md#core

**Reply by AFA141** [reply]
Thanks! I will try

**Comment by AFA141** [comment]
I successfully impletemented TMS tiles, its working just i have one problem. I am getting messages like this: 
 Failed to find directory C:\JavaProjects\mapsforgeAnt\tilestore\13\4558
Application trying to find files/directories that i dont have.. How to fix it?

**Reply by devemux86** [reply]
You need to debug your app.

Also study Mapsforge code to see the messages ([TileStore](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map/src/main/java/org/mapsforge/map/layer/cache/TileStore.java)).

**Reply by AFA141** [reply]
I guess that is not bug, because that message comes up, only when I move/zoom at the part of Container where i dont have tiles, application is trying to load tiles for that part but they are not available and i get messages like this "INFO: Failed to find directory C:\JavaProjects\mapsforgeAnt\tilestore\13\4558."

**Reply by devemux86** [reply]
I changed those log levels to "warning" to not fill the "info" level logs.

---

You can use `MapViewPosition` to set map bounds and zoom limits for existing tiles.

**Reply by AFA141** [reply]
I used already setMapLimit method with corresponding bbox, but that
doesn't prevent INFO logs from being displayed  (in your case Warning logs).

**Reply by devemux86** [reply]
Map limits in map engines work with the bounds at screen center.
The rest of the screen can still request tiles to fill the empty area.

If you are not interested for the logging,
you can turn it off for Mapsforge externally or overwrite the `findFile` method.

**Reply by AFA141** [reply]
Thanks for the help!
